apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsLinkControl.tsx

```diff
@@ -0,0 +1,198 @@
+    icon: BiFileBlank,
+    label: "Page",
+  },
+  external: {
+    icon: BiLink,
+    label: "External link",
+  },
+  file: {
+    icon: BiFile,
+    label: "File",
+  },
+  email: {
+    icon: BiEnvelopeOpen,
+    label: "Email",
+  },
+} as const
+
+export const jsonFormsLinkControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.LinkControl,
+  and(
+    isStringControl,
+    schemaMatches((schema) => schema.format === "link"),
+  ),
+)
+
+const RadioCard = ({ children, ...rest }: PropsWithChildren<UseRadioProps>) => {
+  const { getInputProps, getRadioProps } = useRadio(rest)
+
+  return (
+    <Box
+      as="label"
+      _first={{
+        "> div": {
+          borderLeftRadius: "base",
+        },
+      }}
+      _last={{
+        "> div": {
+          borderRightRadius: "base",
+        },
+      }}
+    >
+      <input {...getInputProps()} />
+      <Box
+        {...getRadioProps()}
+        cursor="pointer"
+        border="1px solid"
+        borderColor="base.divider.strong"
+        bgColor="utility.ui"
+        px="1rem"
+        py="0.5rem"
+        mx={0}
+        _checked={{
+          bgColor: "interaction.muted.main.active",
+          color: "interaction.main.default",
+          borderColor: "interaction.main.default",
+        }}
+        textTransform="none"
+        fontWeight={500}
+        lineHeight="1.25rem"
+      >
+        {children}
+      </Box>
+    </Box>
+  )
+}
+
+interface RadioContentProps {
+  selectedLinkType: string
+  data: string
+  handleChange: (value: string) => void
+}
+
+const RadioContent = ({
+  selectedLinkType,
+  data,
+  handleChange,
+}: RadioContentProps) => {
+  switch (selectedLinkType) {
+    case "page":
+      return (
+        <Input
+          type="text"
+          value={data}
+          onChange={(e) => handleChange(e.target.value)}
+          placeholder="Page permalink"
+        />
+      )
+    case "external":
+      return (
+        <Input
+          type="text"
+          value={data}
+          onChange={(e) => handleChange(e.target.value)}
+          placeholder="https://www.isomer.gov.sg"
+        />
+      )
+    case "file":
+      return (
+        <Input
+          type="text"
+          value={data}
+          onChange={(e) => handleChange(e.target.value)}
+          placeholder="File link"
+        />
+      )
+    case "email":
+      return (
+        <Input
+          type="text"
+          value={data.startsWith("mailto:") ? data.slice("mailto:".length) : ""}
+          onChange={(e) => handleChange(`mailto:${e.target.value}`)}
+          placeholder="test@example.com"
+        />
+      )
+    default:
+      return <></>
+  }
+}
+
+export function JsonFormsLinkControl({
+  data,
+  label,
+  handleChange,
+  path,
+  description,
+  required,
+}: ControlProps) {
+  const [selectedLinkType, setSelectedLinkType] = useState("page")
+
+  const handleLinkTypeChange = (value: string) => {
+    setSelectedLinkType(value)
+    handleChange(path, "")
```

Comment:
does this actually do anything? are you supposed to pass in the change into `handleChange` too?

Reply:
Hmm yeah it is supposed to reset the input field to empty when they change the selected link type. This is because all 4 link types are mutually exclusive of one another, so the input from one does not translate to another, so opted to clear the input field to prevent user confusion.